Realm/Realm/Linq/WhereClauseVisitor.cs

```diff
@@ -0,0 +1,358 @@
+                            Type = GetKind(binaryRightMemberExpression.Type)
+                        };
+                    }
+                    else
+                    {
+                        throw new NotSupportedException(binaryRightMemberExpression + " is null or not a supported node type.");
+                    }
+                }
+
+                if (binaryExpression.Right is ConstantExpression binaryRightConstantExpression)
+                {
+                    if (binaryRightConstantExpression.Value == null)
+                    {
+                        comparisonNode.Right = new ConstantNode()
+                        {
+                            Value = binaryRightConstantExpression.Value,
+                            Type = "string"
+                        };
+                    }
+                    else
+                    {
+                        comparisonNode.Right = new ConstantNode()
+                        {
+                            Value = binaryRightConstantExpression.Value,
+                            Type = GetKind(binaryRightConstantExpression.Value.GetType())
+                        };
+                    }
+                }
+
+                returnNode = comparisonNode;
+            }
+
+            return RealmLinqExpression.Create(returnNode);
+        }
+
+        protected override Expression VisitUnary(UnaryExpression node)
+        {
+            NegationNode returnNode = new NegationNode();
+            switch (node.NodeType)
+            {
+                case ExpressionType.Not:
+                    returnNode.Expression = Extract(node.Operand);
+                    break;
+                default:
+                    throw new NotSupportedException($"The unary operator '{node.NodeType}' is not supported");
+            }
+
+            return RealmLinqExpression.Create(returnNode);
+        }
+
+        protected override Expression VisitMember(MemberExpression node)
+        {
+            if (node.Expression != null && node.Expression.NodeType == ExpressionType.Parameter)
+            {
+                ComparisonNode comparisonNode = new EqualityNode();
+                if (node.Type == typeof(bool))
+                {
+                    comparisonNode.Left = new PropertyNode()
+                    {
+                        Name = GetColumnName(node, node.NodeType),
+                        Type = GetKind(node.Type)
+                    };
+                    comparisonNode.Right = new ConstantNode()
+                    {
+                        Value = true,
+                        Type = "bool"
+                    };
+                }
+
+                return RealmLinqExpression.Create(comparisonNode);
+            }
+
+            throw new NotSupportedException($"The member '{node.Member.Name}' is not supported");
+        }
+
+        private static string GetKind(object valueType)
+        {
+            if (valueType == typeof(float))
```

Comment:
Can we do this with a switch statement? Also, does valueType need to be object? Can we be more specific with the type?

Reply:
Did some research on this and it does not seem like there is a good way to do this, we can discuss this later